docs/data/base/components/modal/modal.md

```diff
@@ -0,0 +1,127 @@
+
+The `Modal` component renders its `children` in front of a backdrop. This lets you create an element that your users must interact with before continuing in the parent application.
+
+Features:
+
+- 🥞 Manages modal stacking when more than one is needed
+- 🔐 Creates a backdrop to disable interaction with the rest of the app
+- 🔐 Disables page scrolling while open
+- ♿️ Manages focus correctly between the modal and its parent app
+- ♿️ Adds the appropriate ARIA roles automatically
+
+{{"component": "modules/components/ComponentLinkHeader.js", "design": false}}
+
+> **Note:** the term "modal" is sometimes used interchangeably with "dialog," but this is incorrect.
+> A dialog may be _modal_ or _nonmodal_.
+>
+> A modal [blocks interaction with the rest of the application](https://en.wikipedia.org/wiki/Modal_window), forcing the user to take action.
+> As such, it should be used sparingly—only when the app _requires_ user input before it can continue.
+
+If you are creating a modal dialog, the [`Dialog`](/material-ui/dialog/) component is better suited for this specific use case.
+
+`Modal` is a lower-level construct that is used in the following components:
+
+- [Dialog](/material-ui/react-dialog/)
+- [Drawer](/material-ui/react-drawer/)
+- [Menu](/material-ui/react-menu/)
+- [Popover](/material-ui/react-popover/)
+
+## Basic modal
+
+```js
+import ModalUnstyled from '@mui/base/ModalUnstyled';
+```
+
+{{"demo": "ModalUnstyled.js", "defaultCodeOpen": false }}
+
+## Nested modal
+
+> ⚠ **Note:** though it is possible to create nested modals—for example, a select modal within a dialog—stacking more than two at a time is discouraged.
+
+{{"demo": "NestedModal.js", "defaultCodeOpen": false}}
+
+## Transitions
+
+You can animate the open and close states of a modal with a transition component, as long as that component fulfills the following requirements:
+
+- Is a direct child descendent of the modal
+- Has an `in` prop—this corresponds to the open/close state
+- Calls the `onEnter` callback prop when the enter transition starts
+- Calls the `onExited` callback prop when the exit transition is completed
+  > The `onEnter` and `onExited` callbacks tell the modal to unmount the child content when closed and fully transitioned.
+
+`Modal` has built-in support for [react-transition-group](https://github.com/reactjs/react-transition-group):
+
+{{"demo": "TransitionsModal.js", "defaultCodeOpen": false}}
+
+You can also use [react-spring](https://github.com/pmndrs/react-spring) with the `Modal`, but it will require additional custom configuration:
+
+{{"demo": "SpringModal.js", "defaultCodeOpen": false}}
+
+## Performance
+
+The Modal component's content is unmounted when it is not opened.
+This means that it will need to be re-mounted each time it is _opened_. If you are rendering expensive component trees inside your modal, and you want to optimize for interaction responsiveness, you can change this default behavior by enabling the keepMounted prop.
```

Comment:
I should have been more specific here @mnajdova—"opened" doesn't need to be italicized, I just added that in my earlier comment to emphasize what I had changed. :)

```suggestion
This means that it will need to be re-mounted each time it is opened. If you are rendering expensive component trees inside your modal, and you want to optimize for interaction responsiveness, you can change this default behavior by enabling the keepMounted prop.
```